onnxruntime/core/graph/model.cc

```diff
@@ -348,10 +348,27 @@ Status Model::Load(int fd, std::shared_ptr<Model>& p_model, const IOnnxRuntimeOp
   if (fd < 0) {
     return Status(ONNXRUNTIME, INVALID_ARGUMENT, "<p_fd> less than 0.");
   }
+#if GOOGLE_PROTOBUF_VERSION >= 3002000
   std::unique_ptr<ModelProto> model_proto = std::make_unique<ModelProto>();
   if (!model_proto->ParseFromFileDescriptor(fd)) {
-    return Status(ONNXRUNTIME, INVALID_PROTOBUF, "Protobuf parsing failed.");
+      return Status(ONNXRUNTIME, INVALID_PROTOBUF, "Protobuf parsing failed.");
+  }
+#else
+  auto raw_input = std::unique_ptr<ZeroCopyInputStream>(std::make_unique<FileInputStream>(fd));
```

Comment:
Indeed, these two vars doesn't need to be allocated on stack.
You can write:
FileInputStream fs(fd);
CodedInputStream cis(&fs);

Reply:
thanks, let me verify

Reply:
Good catch, it works. Thanks!